app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,37 @@
+class FoodsController < ApplicationController
+  load_and_authorize_resource
+
+  def index
+    @user = current_user
+    @foods = current_user.foods.all
+  end
+
+  def show
+    @food = Food.find(params[:id])
+  end
+
+  def new
+    @food = current_user.foods.new
+  end
+
+  def create
+    @food = current_user.foods.create(food_params)
+    if @food.save
+      redirect_to user_foods_path
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @food = current_user.foods.find(params[:id])
+    @food.destroy
```

Comment:
- [ ] When I try to delete a food item, it raises a foreign key violation error. Kindly fix this issue. I think modifying the `has_many :recipe_foods` association to `has_many :recipe_foods, dependent: :destroy` would solve the issue. :pray: 

![Screenshot from 2021-12-10 16-23-04](https://user-images.githubusercontent.com/64894927/145581024-322917f0-92ef-4aff-aac2-11967e114404.png)